// @duckdb/node-api@1.5.0-r.1/lib/DuckDBBindInfo.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DuckDBBindInfo = void 0;
const node_bindings_1 = __importDefault(require("@duckdb/node-bindings"));
const DuckDBClientContext_1 = require("./DuckDBClientContext");
class DuckDBBindInfo {
    bind_info;
    constructor(bind_info) {
        this.bind_info = bind_info;
    }
    get clientContext() {
        return this.getClientContext();
    }
    getClientContext() {
        return new DuckDBClientContext_1.DuckDBClientContext(node_bindings_1.default.scalar_function_get_client_context(this.bind_info));
    }
    get extraInfo() {
        return this.getExtraInfo();
    }
    getExtraInfo() {
        return node_bindings_1.default.scalar_function_bind_get_extra_info(this.bind_info);
    }
    setBindData(bindData) {
        node_bindings_1.default.scalar_function_set_bind_data(this.bind_info, bindData);
    }
    setError(error) {
        node_bindings_1.default.scalar_function_bind_set_error(this.bind_info, error);
    }
}
exports.DuckDBBindInfo = DuckDBBindInfo;
